packages/server/src/modules/product/ProductModel.ts

```diff
@@ -0,0 +1,50 @@
+import mongoose, { Document, Model } from 'mongoose';
+
+const { ObjectId } = mongoose.Schema.Types;
+const Schema = mongoose.Schema;
+
+const ProductSchema = new Schema(
```

Comment:
declaring the interface before you can use with schema like `new Schema< IProduct>(`